web-console/knikubevirt/VMIs/VMIs.md

```diff
@@ -0,0 +1,60 @@
+User can delete VMs and orphan the underlying VMIs so that the VM will get deleted but the VMI will still be kept.
+2. Users can write controllers for VMIs if they need specific UCs which aren’t covered by VMs. F.e. Users would never see VMIs because they are not shown.
+3. If in K8s we create Pods via Deployment, then why not create VMI via VMs so the user will learn K8s via VMs? Why create a gap between virtualization and K8s?
+4. Upstream Kubernetes is popular because in VMI Replica set and machine controller implementation that is used to “spork(?)” Kubernetes clusters on top of CNV
+Bare metal machines installed with Kubernetes
+Users install Kubevirt for CNV
+On these VMs there are clusters that can be scaled up and down and they create the VMIs (and not the VMs)
+
+## 2 Types of personas
+
+1. Traditional VM operation admin that is used to a certain view and behavior of the console, in a traditional virtualization way/ A user who has a virtualization background and has no need to understand the underlying k8s technology
+2. Users that use the console in a more ‘Kubernetes’ manner
+We need to try to educate users and narrow the gap for CNV users to better understand K8s/OpenShift.
+We want to allow this to be surfaced in the UI and live side by side in a way that will not confuse/ frighten/ intimidate the traditional user.
+
+## Mockups
+
+We had finalized into this option and will show 2 parallel columns in the VMs list page, one for each: VMs beside VMIs.
+
+![list of VMIs beside VMs](img/VMsListW_VMIsOp1.png)
+
+Trying to educate users about VMIs, we will provide hints, whenever it is appropriate. 
+In most usual cases, when no VM or VMI is missing, the instance name link will take the user directly to the instance page (VM/VMI).
+When a VMI is missing an owner VM, the user will be provided with an explanation in a popover for the reason it is missing. The name of the VM + badge will be shown in a disabled state for the sake of accessibility so a screen reader would be able to announce it.
+
+![VMI is in an off status](img/Op1_Hint1.png)
+
+### UI definitions for missing VMs and VMIs in the table in a disabled state
+
+Color of badge: #737679, background color of badge: #D2D2D2, color of name text: #737679
+
+If a VMI is in an off status, the user will also be notified about that via a popover.
+
+![Missing VMI](img/Op1_PopoverMessageMissingVMI.png)
+
+## VMI's details view
+
+VMI's details page should be similar to the VM’s details page and act in a complementary manner.
+The VMI's details page will provide a hint with a general explanation about what are VMIs and also allow drill down to further details.
+
+![VMI's details page](img/Op1_VMI_DetailsViewPlusHint.png)
```

Comment:
> Does it mean that any edits they make to the VMI on this page will be overwritten by the VM:

Kobi explains that VMIs Like Pods will change if their owner changes, so if this VMI has an owner (VM or VMIRS), they will change on the next boot to reconcile with their owner.

> Do the actions in the Actions menu on this page apply to the VM instead of the VMI?
No, these actions apply to the VMI's.

Reply:
+1 @lizsurette thanks!